app/assets/javascripts/controllers/middleware_server/middleware_server_controller.js

```diff
@@ -0,0 +1,149 @@
+      $scope.paramsModel.operationButtonName = makeOperationDisplayName(operation);
+    }
+    $scope.paramsModel.timeout = 10; // default timeout value
+  };
+
+  $scope.runOperation = function () {
+    $scope.$broadcast('mwSeverOpsEvent', $scope.paramsModel);
+  };
+
+  var makeOperationDisplayName = function (operation) {
+    return _.capitalize(operation);
+  };
+
+  /////////////////////////////////////////////////////////////////////////
+  // Add Deployment
+  /////////////////////////////////////////////////////////////////////////
+
+  $scope.deployAddModel = {};
+  $scope.deployAddModel.enableDeployment = true;
+  $scope.deployAddModel.serverId = angular.element('#server_id').val();
+
+  $scope.showDeployListener = function () {
+    $scope.deployAddModel.showDeployModal = true;
+    $scope.resetDeployForm();
+  };
+
+  $scope.resetDeployForm = function () {
+    $scope.enableDeployment = true;
+    $scope.runtimeName = undefined;
+    $scope.filePath = undefined;
+    angular.element('#deploy_div :file#upload_file').val('');
+    angular.element('#deploy_div input[type="text"]:disabled').val('');
+  };
+
+  $scope.$watch('filePath', function(newValue) {
+    if (newValue) {
+      $scope.deployAddModel.runtimeName = newValue.name;
+    }
+  });
+
+  $scope.addDeployment = function () {
+    miqService.sparkleOn();
+    $scope.$broadcast('mwAddDeploymentEvent', $scope.deployAddModel);
+  };
+}
+
+ManageIQ.angular.app.controller('mwServerOpsController', MwServerOpsController);
+
+MwServerOpsController.$inject = ['$scope', 'miqService', 'serverOpsService'];
+
+function MwServerOpsController($scope, miqService, serverOpsService) {
+
+  $scope.$on('mwSeverOpsEvent', function(event, data) {
+    miqService.sparkleOn();
+
+    serverOpsService.runOperation(data.serverId, data.operation, data.timeout)
+      .then(function (response) {
+          miqService.miqFlash('success', response);
+        },
+        function (error) {
+          miqService.miqFlash('error', error);
+
+        }).finally(function () {
+
+      miqService.sparkleOff();
+    });
+  });
+}
+
+ManageIQ.angular.app.service('serverOpsService', ServerOpsService);
+
+ServerOpsService.$inject = ['$http', '$q'];
+
+function ServerOpsService($http, $q) {
+  this.runOperation = function runOperation(id, operation, timeout) {
+    var errorMsg = _('Error running operation on this server.');
+    var deferred = $q.defer();
+    var payload = {
+      'id': id,
+      'operation': operation,
+      'timeout': timeout
+    };
+
+    $http.post('/middleware_server/run_operation', angular.toJson(payload))
+      .then(
+        function (response) { // success
+          var data = response.data;
+
+          if (data.status === 'ok') {
+            deferred.resolve(data.msg);
+          } else {
+            deferred.reject(data.msg);
+          }
+        },
+        function () { // error
```

Comment:
An effort to keep the controllers as thin as possible and move as much as possible to the services.